controller/src/main/java/org/jboss/as/controller/operations/global/ReadResourceHandler.java

```diff
@@ -516,7 +524,14 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
                     PathElement path = entry.getKey();
                     ModelNode value = entry.getValue();
                     iter.remove();
+                   // ModelNode responseHeaders = entry.getValue().get(RESPONSE_HEADERS).get(WARNINGS);
                     if (!value.has(FAILURE_DESCRIPTION)) {
+                        if (value.hasDefined(RESPONSE_HEADERS, WARNINGS)){
+                            for (ModelNode response : value.get(RESPONSE_HEADERS).get(WARNINGS).asList()) {
+                                String level = response.get("level").toString().replace("\"", "");
```

Comment:
You have to use `response.get("level").asString()` instead of `response.get("level").toString().replace("\"", "");`

In addition, it is always preferred to get the model attribute name from the available `ModelDescriptionConstants` class.  Instead of `response.get("level")` you should use `response.get(ModelDescriptionConstants.LEVEL)`